dev/io.openliberty.microprofile.telemetry.2.0.internal/src/io/openliberty/microprofile/telemetry20/internal/config/OpenTelemetryVersionedConfigurationImpl.java

```diff
@@ -9,36 +9,127 @@
 
 import org.osgi.service.component.annotations.Component;
 
-import io.openliberty.microprofile.telemetry.internal.common.info.OpenTelemetryInfoFactoryImpl;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+import io.openliberty.microprofile.telemetry.internal.common.AgentDetection;
 import io.openliberty.microprofile.telemetry.internal.common.constants.OpenTelemetryConstants;
+import io.openliberty.microprofile.telemetry.internal.common.info.OpenTelemetryInfoFactoryImpl;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Classes;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Cpu;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.GarbageCollector;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.MemoryPools;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Threads;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
 import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
-import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.resources.ResourceBuilder;
 
 /**
  * This class contains version specific configuration for OpenTelemetryInfoFactory
  */
 @Component
 public class OpenTelemetryVersionedConfigurationImpl implements OpenTelemetryInfoFactoryImpl.OpenTelemetryVersionedConfiguration {
 
+    private static final TraceComponent tc = Tr.register(OpenTelemetryVersionedConfigurationImpl.class);
+
     // Version specific API calls to AutoConfiguredOpenTelemetrySdk.builder()
     @Override
-    public AutoConfiguredOpenTelemetrySdkBuilder getPartiallyConfiguredOpenTelemetrySDKBuilder() {
-        return AutoConfiguredOpenTelemetrySdk.builder()
-                        .disableShutdownHook();
+    public OpenTelemetry buildOpenTelemetry(Map<String, String> openTelemetryProperties,
+                                            BiFunction<? super Resource, ConfigProperties, ? extends Resource> resourceCustomiser, ClassLoader classLoader) {
+
+        openTelemetryProperties.putAll(getTelemetryPropertyDefaults());
+
+        OpenTelemetrySdk openTelemetry = AutoConfiguredOpenTelemetrySdk.builder()
+                        .addPropertiesCustomizer(x -> openTelemetryProperties) //Overrides OpenTelemetry's property order
+                        .addResourceCustomizer(resourceCustomiser) //Defaults service name to application name
+                        .setServiceClassLoader(classLoader)
+                        .disableShutdownHook()
+                        .build()
+                        .getOpenTelemetrySdk();
+
+        // Register observers for runtime metrics
+        Classes.registerObservers(openTelemetry);
```

Comment:
Do we need to check for JDKs that are not hotspot or J9 and not registering the JVM metrics?

Reply:
We do, its on my list.